### PR TITLE
8.8 what's new

### PR DESCRIPTION
**Pull request by Pessimistress**
For #6967 

#### Change List
- What's new
- Upgrade guide

### CODE REVIEW

File: docs/whats-new.md [hunk 1]
```diff
@@ -2,6 +2,35 @@
...
+
+### MapboxOverlay
+
+The `@deck.gl/mapbox` module now exports a new class [MapboxOverlay](/docs/api-reference/mapbox/mapbox-overlay.md). The class implements Mapbox GL JS's [IControl](https://docs.mapbox.com/mapbox-gl-js/api/markers/#icontrol) API. When adding a `MapboxOverlay` control to an mapbox map, deck.gl layers are rendered in synchronization with the base map layers. This control supports both [overlaid and interleaved](/docs/get-started/using-with-map.md) rendering modes. See the new [get started example](/examples/get-started/pure-js/mapbox/).
+
+If you are using react-map-gl v7, this is the only solution to use deck.gl with React map control components (`Navigationcontrol`, `GeolocateControl`, etc.). See [example](https://github.com/visgl/react-map-gl/blob/master/examples/deckgl-overlay/src/app.tsx).
+
+### `@deck.gl/carto`
+
+TODO
```
Comment (Pessimistress) [inline docs/whats-new.md:31]: @alasarr

File: docs/whats-new.md [hunk 2]
```diff
@@ -2,6 +2,35 @@
...
 
+## deck.gl v8.8
+
+Release date: June, 2022
+
+### TypeScript
+
+Official typings are now available in deck.gl for public preview. It was a tremendous effort converting the deck.gl code base into TypeScript. We did it not just for TypeScript users to easily consume this library, but also for the overall robustness and maintainability of our own code base.
+
+In v8.x, the types will not be exposed by default to prevent any typing errors from breaking existing TypeScript applications. You can opt in to using the typed version by switching your import statements to the `@deck.gl/*/typed` endpoints. Visit [getting started with TypeScript](/docs/get-started/using-with-typescript.md) to find out more.
```
Comment (felixpalmer) [inline docs/whats-new.md:13]: Do we want to give an indication as to when we plan on exposing them by default? To give advance warning and to encourage people to try them

**Review by ibgreen — APPROVED** [review]
While long explanations can be  useful as background, Please make an effort start with a list of one liner bullets summarizing the changes. Use "declarative" language stating what was changed, not why.

Also is possible run a spell checker on the text there are multiple obvious typos.

File: docs/whats-new.md [hunk 3]
```diff
@@ -2,6 +2,35 @@
...
+
+### Custom Indexing System in TileLayer
+
+The [TileLayer](/docs/api-reference/geo-layers/tile-layer.md) is foundamental when it comes to visualizing datasets so big that they cannot fit in the browser's memory. The idea is to only fetch and render content that is visible in the current viewport, and at the appropriate detail level. Up till now, the `TileLayer` has exclusivel implemented [the OSM tile index](https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames). If you wish to use it with your dataset, you had to use an offline tool or an backend that partitioned geospatial data into OSM tiles.
```
Comment (kylebarron) [inline docs/whats-new.md:17]:
```suggestion
The [TileLayer](/docs/api-reference/geo-layers/tile-layer.md) is fundamental when it comes to visualizing datasets so big that they cannot fit in the browser's memory. The idea is to only fetch and render content that is visible in the current viewport, and at the appropriate detail level. Up till now, the `TileLayer` has exclusively implemented [the OSM tile index](https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames). If you wish to use it with your dataset, you had to use an offline tool or an backend that partitioned geospatial data into OSM tiles.
```

**Review by kylebarron — APPROVED** [review]

File: docs/whats-new.md [hunk 4]
```diff
@@ -2,6 +2,35 @@
...
+
+The `@deck.gl/mapbox` module now exports a new class [MapboxOverlay](/docs/api-reference/mapbox/mapbox-overlay.md). The class implements Mapbox GL JS's [IControl](https://docs.mapbox.com/mapbox-gl-js/api/markers/#icontrol) API. When adding a `MapboxOverlay` control to an mapbox map, deck.gl layers are rendered in synchronization with the base map layers. This control supports both [overlaid and interleaved](/docs/get-started/using-with-map.md) rendering modes. See the new [get started example](/examples/get-started/pure-js/mapbox/).
+
+If you are using react-map-gl v7, this is the only solution to use deck.gl with React map control components (`Navigationcontrol`, `GeolocateControl`, etc.). See [example](https://github.com/visgl/react-map-gl/blob/master/examples/deckgl-overlay/src/app.tsx).
+
+### `@deck.gl/carto`
+
+TODO
+
+
```
Comment (alasarr) [inline docs/whats-new.md:33]: Need to add Geohash Layer

File: docs/whats-new.md [hunk 5]
```diff
@@ -2,6 +2,43 @@
...
+Due to this generalization, there is a breaking change affecting indexing properties (`x`, `y` and `z`) in the `TileLayer` API. See [upgrade guide](/docs/upgrade-guide.md) for details.
+
+### MapboxOverlay
+
+The `@deck.gl/mapbox` module now exports a new class [MapboxOverlay](/docs/api-reference/mapbox/mapbox-overlay.md). The class implements Mapbox GL JS's [IControl](https://docs.mapbox.com/mapbox-gl-js/api/markers/#icontrol) API. When adding a `MapboxOverlay` control to an mapbox map, deck.gl layers are rendered in synchronization with the base map layers. This control supports both [overlaid and interleaved](/docs/get-started/using-with-map.md) rendering modes. See the new [get started example](/examples/get-started/pure-js/mapbox/).
+
+If you are using react-map-gl v7, this is the only solution to use deck.gl with React map control components (`Navigationcontrol`, `GeolocateControl`, etc.). See [example](https://github.com/visgl/react-map-gl/blob/master/examples/deckgl-overlay/src/app.tsx).
+
+### `@deck.gl/carto`
```
Comment (alasarr) [inline docs/whats-new.md:29]: I've added this, but we need to revisit it next week

- We're going to create examples at CARTO doc that I'd like to link from the module doc.

- A link to QuadBin to explain this XYZ index better

**Review by alasarr — COMMENTED** [review]
Looks good to me, just missing the Geohash Layer

File: docs/whats-new.md [hunk 6]
```diff
@@ -2,6 +2,43 @@
...
+
+- CartoLayer includes native support to work with spatial indexes like [H3](https://h3geo.org/) or QuadBin. 
+
+```js
```
Comment (Pessimistress) [inline docs/whats-new.md:33]: Remove code sample from what's new. You can link to the documentation page.

**Pull request closed without merging**